### PR TITLE
Filter locale on REST context

### PR DESCRIPTION
**Pull request by Marianne380**
Fix https://github.com/polylang/polylang-wc/issues/787.

Filter locale on **REST** context to set the right current language user meta.

**_Needs https://github.com/polylang/polylang/pull/1395 to work because of `do_action( 'change_locale', $new_locale );` which reload the WC localisation files._**

### CODE REVIEW

File: include/rest-request.php [hunk 1]
```diff
@@ -122,4 +124,20 @@ public function set_language( $result, $server, $request ) {
...
+	 *
+	 * @param string $locale The current ID.
+	 * @return string
+	 */
+	public function get_locale( $locale ) {
+		if ( ! empty( $this->curlang ) ) {
+			return $this->curlang->locale;
+		}
+
+		return $locale;
```
Comment (szepeviktor) [inline include/rest-request.php:141]:
```suggestion
		if ( empty( $this->curlang ) ) {
			return $locale;
		}

		return $this->curlang->locale;
```

Please put the unhappy path in the conditional.

[_source_](https://github.com/szepeviktor/very-clean-code#-happy-and-unhappy-path)

**Comment by szepeviktor** [comment]
@Chouby PHPStan says `$this` is of type mixed because `PLL_Static_Pages` has constructor with a pass-by-ref (`&$polylang`) parameter. So `PLL_Static_Pages` may modify `$this`.

**Review by Hug0-Drelon — APPROVED** [review]
LGTM, but I think it'd be worthy to add some tests. What do you think?

**Comment by Marianne380** [comment]
Replaced by https://github.com/polylang/polylang-wc/pull/791.

**Pull request closed without merging**